components/automate-cs-ocid/habitat/config/tasks/save_secret_key_base_file.rb

```diff
@@ -0,0 +1,67 @@
+require 'base64'
+
+class SecretKeyBaseGenerator
+  # This constant will be used to extract the first `n` characters from the erchef webui private key
+  # which will be encrypted to generate the secret_key_base for OCID rails app.
+  WEBUI_KEY_SUBSTRING_LEN = 75
+  class << self
+    # We will save the secret_key_base in a file which will later be used
+    # to set it in the OCID environment configuration
+    def save_secret_key_base_in_file
+      begin
+        custom_secret_key_base = ENV['CUSTOM_SECRET_KEY_BASE']
+
+        # If user has provided any custom secret_key_base in automate configuration
+        # it will be considered, otherwise we will generate a secret_key_base from the erchef webui_key
+        secret_key_base = if (custom_secret_key_base.nil? || custom_secret_key_base.empty?)
+          generate_secret_key_base
+        else
+          custom_secret_key_base
+        end
+        
+        secret_key_base_file_path = ENV['SECRET_KEY_BASE_FILE_PATH']
+        
+        if File.exists?(secret_key_base_file_path)
+          puts "SECRET_KEY_BASE file exists. File will be overwritten.."
+        else
+          puts "SECRET_KEY_BASE file doesn't exist. It will be generated.."
+          dir = File.dirname(secret_key_base_file_path)
+          unless File.directory?(dir)
+            FileUtils.mkdir_p(dir)
+          end
+        end
```

Comment:
@RajeshPaul38 This can be extracted out as a separate function. Example - `get_dir`